src/components/screens/admin-vacation-management/EditVacationDialog.tsx

```diff
@@ -67,42 +85,84 @@ const EditVacationDialog = ({
   onSave,
   disableSave
 }: EditVacationDialogProps) => {
+  const currentYear = new Date().getFullYear().toString();
+  const[selectedYear,setSelectedYear] = useState<string>(currentYear);
+  const availableYears = generateYearOptions();
+
   if (!user) return null;
 
   const normalizedVacationDays = normalizeVacationDays(vacationDays);
-  const years = Object.keys(normalizedVacationDays).filter(
-    (year) => Number.parseInt(year) <= new Date().getFullYear()
-  );
+
+  // Get the data for the selected year, or initialize with zero
+  const selectedYearData = normalizedVacationDays[selectedYear] || { total: 0, remaining: 0 };
+
+  const handleYearChange = (year: string) => {
```

Comment:
However as a general rule all functions should have TSDocs, with a one line descriptuion and @param mark for any params used